### PR TITLE
feat(playback): resume non-shuffle position & restart playlist (#786)

### LINKED ISSUE #786: Persist & resume non-shuffle playback position + Restart playlist action
## Summary

Frames on scheduled power restart at photo #1 every cycle because the non-shuffle playlist rebuilds from index 0 on startup. Persist the last **non-shuffle** displayed media id and resume there on the next non-shuffle `build_playlist`, and add a "Restart playlist" action as the deliberate reset path.

**Related to:** #785
**Branch:** `feat/785-resume-non-shuffle-playback` (from `dev`)

## Why media id (not raw index)
A raw index silently drifts to the wrong photo when the media monitor adds/removes files or `sort_cols` changes. An id match lands on the correct slot or safely falls back to 0.

## Why a dedicated marker (not the global `last_displayed`)
`last_displayed` is mode-agnostic; reusing it would resume at a shuffled slot when switching shuffle -> non-shuffle. The marker is written **only when `shuffle is False`** and read **only when `shuffle is False`**, so it always reflects the sequential run position.

## UX placement (decided)
| Surface | Decision | Why |
|---|---|---|
| Remote | **In the shuffle dropdown** as a divided action row, adaptive label | Playlist-ordering action; right category; no grid disruption |
| Appearance | No | It is an action, not a durable setting |
| Dock inline | No | Not a primary transport; would clutter a minimal touch deck |
| Dock danger menu | No | Not system-destructive; confirm modal is over-protective |

The action label adapts by mode (one underlying `RESTART_PLAYLIST` command):
- Non-shuffle -> **"Restart playlist"** (de: "Wiedergabeliste neu starten")
- Shuffle -> **"Reshuffle now"** (de: "Jetzt neu mischen")

## Backend
- `core/services/playlist.py` (`PlaylistManager`):
  - `_persist_resume_media_id(media_id)` / `_resume_playlist_position()` helpers using `config_repo.get_app_config`/`set_app_config`, key `model.resume_media_id`. No-op when `_config_repo is None`.
  - `build_playlist()`: after building `_display_playlist`, if `not self._shuffle` -> resume (find first slot containing stored id, `_current_index = (slot_idx + 1) % len`; not found / no marker -> 0); else keep `_current_index = 0`.
  - `_prepare_slot_for_display()`: after successful `record_media_displayed`, if `not self._shuffle` -> persist slot's first media id.
  - `restart_playlist() -> None`: clear `model.resume_media_id`, `build_playlist(shuffle=self._shuffle)`, reset `_run_through_count = 0`.
- `core/events/dto.py`: add `RESTART_PLAYLIST = auto()` to `Command`.
- `core/engine/playback.py` (`_handle_command`): handle `Command.RESTART_PLAYLIST` -> `restart_playlist()` + `trigger next media`.

## API
- `api/models.py` (`WebSocketCommandMessage.command`): add `"RESTART_PLAYLIST"` to the `Literal[...]`.
- `api/app.py` (WebSocket dispatch, near `STOP`): publish `CommandEvent(Command.RESTART_PLAYLIST)`.

## Frontend
- `frontend/src/stores/player.ts`: `function restartPlaylist() { sendCommand('RESTART_PLAYLIST') }` (mirror `next`/`previous`); export it.
- `frontend/src/views/RemoteView.vue`: add a divided action row at the bottom of the shuffle dropdown; label adapts from `playerStore` shuffle state; wire `@click="playerStore.restartPlaylist()"`.
- `frontend/src/locales/en.json` + `de.json` (kept in sync): `remote.controls.restartPlaylist` ("Restart playlist" / "Wiedergabeliste neu starten"), `remote.controls.reshuffleNow` ("Reshuffle now" / "Jetzt neu mischen").

## Tests (TDD)
- `test/core/services/test_playlist.py`: resume finds slot +1; id absent / no marker / no config repo -> 0; marker written only when `shuffle=False`; shuffle build skips read; `restart_playlist()` clears marker + rebuilds at 0; shuffle `restart_playlist` rebuilds fresh shuffle; portrait-pair slot resumes by either contained id.
- `test/core/engine/test_playback.py`: `Command.RESTART_PLAYLIST` calls `restart_playlist()` + triggers next.
- `test/api/test_app.py`: WebSocket `RESTART_PLAYLIST` publishes `CommandEvent`.
- Frontend `player` store test: `restartPlaylist()` sends `'RESTART_PLAYLIST'`.

## Edge cases
- Resume only on `build_playlist` (startup + `model` config change); loop exhaustion via `_advance_playlist_cycle` resets to 0 without calling `build_playlist` -> no fight with the daily loop.
- Shuffle on -> never read/write marker; toggling off->on->off keeps last sequential position.
- Filter / `pic_dir` / `sort_cols` change or file deleted -> id not found -> start at 0.
- Fresh install / cleared `config.db3` -> no marker -> start at 0.
- `config_repo is None` -> helpers no-op (today's behavior).
- New files added before the resume point -> id still found; earlier new files shown next loop.

## Verification
- `.venv/bin/python -m pytest`
- `.venv/bin/python -m mypy src/picframe`
- `.venv/bin/python -m ruff check .` + `ruff format --check .`
- `cd frontend && yarn test && yarn lint && yarn format:check && yarn build`

## Docs / memory bank
- `docs/user/manual.md`: note resume-on-restart (non-shuffle) + Restart playlist action.
- `memory-bank/systemPatterns.md` + `activeContext.md`: record non-shuffle resume via `model.resume_media_id` (non-shuffle-only marker) + `RESTART_PLAYLIST` command.

## Acceptance
- [ ] Non-shuffle frame restart resumes at the last shown photo (+1).
- [ ] Shuffle never persists/reads the marker.
- [ ] Restart playlist action clears marker (non-shuffle -> 0) / reshuffles (shuffle).
- [ ] Action lives in the Remote shuffle dropdown, adaptive label.
- [ ] All quality gates green.

### PR DESCRIPTION
**Pull request by helgeerbe**
## Summary

Closes #786.

Implements persistence and resumption of the non-shuffle playback position, plus a dedicated "Restart playlist" command.

### Resume marker (non-shuffle only)
`PlaylistManager` persists the primary media id of each displayed slot to `model.resume_media_id` in `config.db3` **only while `shuffle is False`** (shuffle order is not stable across restarts, so a marker would be meaningless). On the first non-shuffle `build_playlist` after startup, `_apply_resume_position()` advances the cursor past the marked slot (slot+1; wraps to 0 at the end; falls back to 0 if the marked item is gone). One-shot via `_resume_applied` so config-change rebuilds and `restart_playlist` don't re-jump. Robust against file additions/reordering because it keys on media id, not raw index.

### Restart playlist command
New `Command.RESTART_PLAYLIST` (high priority) handled in `PlaybackEngine._handle_restart_playlist_command` → `PlaylistManager.restart_playlist()` (clears the marker, drops history, rebuilds with the current shuffle setting) then `_trigger_next_media()`. WebSocket dispatch added in `api/app.py`; `"RESTART_PLAYLIST"` added to the `WebSocketCommandMessage` literal in `api/models.py`.

### UI
The Remote shuffle dropdown gained a divided action row with an adaptive label — "Restart playlist" (shuffle off) / "Reshuffle now" (shuffle on) — one underlying command (`player.ts` `restartPlaylist()`, `RemoteView.vue` action button, i18n `restartPlaylist`/`reshuffleNow` en+de).

### Tests
- `test_playlist.py` — resume +1, fallback, shuffle-skip, persist on/off, restart clears
- `test_playback.py` — RESTART_PLAYLIST dispatch
- `test_app.py` — WS→CommandEvent
- `player.test.ts` — WS send

All gates green: `pytest` (931 in core+api) / `mypy` / `ruff`; `yarn test` (142) / `lint` / `format` / `build` (vue-tsc). Docs: `manual.md`.

## Commits
- `feca7d5` feat(playlist): persist & resume non-shuffle playback position (#786)
- `31b7769` feat(playback): add RESTART_PLAYLIST command (#786)
- `063aff2` feat(remote): add restart/reshuffle action to shuffle dropdown (#786)
- `ef05edc` docs(manual): document resume & restart playlist (#786)

## Summary by Sourcery

Preserve non-shuffled playback progress and provide a unified command to restart or reshuffle the current playlist.

New Features:
- Persist and resume non-shuffled playback from the item following the last displayed media.
- Add a high-priority restart playlist command that resets playback and reshuffles when shuffle is enabled.
- Expose restart and reshuffle actions in the Remote shuffle menu with localized adaptive labels.

Enhancements:
- Use media identifiers for robust resume behavior across playlist changes and avoid resuming shuffled playback.

Documentation:
- Document non-shuffled playback resumption and the restart/reshuffle action.

Tests:
- Add backend, WebSocket, and frontend coverage for resume behavior and restart playlist command dispatch.

### CODE REVIEW

File: src/picframe/core/services/playlist.py [hunk 1]
```diff
@@ -122,10 +127,12 @@ def build_playlist(self, shuffle: bool | None = None) -> None:
         if not self._display_playlist:
             logger.warning("No media items found to build playlist.")
             self._current_index = -1
+            self._resume_applied = True
```
[the remark this answers is not in the document]
Comment (helgeerbe) [reply src/picframe/core/services/playlist.py:130]: Reviewed — the finding accurately describes the code, but the behavior is **by design**, not a bug:

- The `_resume_applied` latch is a **startup-only, one-shot** mechanism. It is intentionally consumed on the first build after startup regardless of shuffle mode or an empty playlist, so resume only fires when Picframe starts in non-shuffle mode with media present.
- Toggling shuffle off mid-session is expected to be a **clean playlist restart from slot 0** (the config-change rebuild path resets `_current_index = 0` and the spent latch prevents resume). Preserving position across a shuffle toggle is out of scope for #786 and would be a separate feature.
- No data loss: the marker stays in `config.db3`; worst case is starting at slot 0 for one session.

Rather than change the code (which would create an inconsistency where a fresh start that begins in shuffle and then switches would resume, but a mid-session toggle would not), we aligned the **docstring/comments** with the actual behavior in #789. The misleading *"applied once, on the first non-shuffle build after startup"* wording is corrected to state the latch is consumed on the first build of any kind.

Resolving as by-design.
